featuretools/utils/trie.py

```diff
@@ -0,0 +1,116 @@
+class Trie(object):
+    """
+    A trie (prefix tree) where the keys are sequences of hashable objects.
+
+    It behaves similarly to a dictionary, except that the keys can be lists or
+    other sequences.
+
+    Examples:
+        .. code-block:: python
+
+            from featuretools.utils import Trie
+
+            trie = Trie(default=str)
+
+            # Set a value
+            trie[[1, 2, 3]] = '123'
+
+            # Get a value
+            assert trie[[1, 2, 3]] == '123'
+
+            # Overwrite a value
+            trie[[1, 2, 3]] = 'updated'
+            assert trie[[1, 2, 3]] == 'updated'
+
+            # Getting a key that has not been set returns the default value.
+            assert trie[[1, 2]] == ''
+    """
+    def __init__(self, default=lambda: None, path_constructor=list):
+        """
+        default: A function returning the value to use for new nodes.
+        path_constructor: A function which constructs a path from a list. The
+            path type must support addition (concatenation).
+        """
+        self._value = default()
+        self._children = {}
+        self._default = default
+        self._path_constructor = path_constructor
+
+    @property
+    def value(self):
```

Comment:
is there any reason to have these defined using `@property`? I think we can just have `self.value` defined